### DISCUSSION TITLE
멀티플레이 중 상호작용 버튼을 가까이 다가간 플레이어만 볼 수 있게 하는 방법이 있을까요?

### DISCUSSION THREAD

**Discussion by syjeuion** [naverz/zepeto-studio-kor#870] [5. Q&A]
멀티플레이 게임을 제작 중인데요,
공식 월드의 상호작용 버튼처럼 가까이 다가간 플레이어만 버튼이 보이고 주위의 다른 플레이어는 계속 안 보이게 하고 싶습니다.

현재 collider를 이용해서 ontriggerstay 상황일때 버튼이 활성화 되게
이런 식의 접근을 해보았으나
플레이어1이 접근을 하면 멀리 있는 플레이어2한테도 버튼이 보이더라고요.

1. 특정 플레이어 한테만 특정 오브젝트가 보이게 하는 방법
or
2. 가까이 다가간 플레이어 한테만 활성화 되는 방법

이런 식의 구현을 원하는데 어떻게 조건을 걸어도 조건이 충족되면 해당 서버에 있는 플레이어 전체에게 보이기에 질문 드립니다.

layer를 이용하는 방법도 고려 중이나, 다른 레이어에 있는 플레이어가 접근했을 때 
활성화가 안되는건 가능하지만 보이지 않게 하는건 안되더라고요.
혹시 방법을 아신다면/ 이런 방법으로 하면 되지 않을까? 싶은 방법이 떠오르시면 알려주시면 감사하겠습니다.

**Comment by YSHFM** [comment]
저는2가지 방법이 생각나는데,
1번째로는 플레이어가 접촉 시 접촉한 순간 접촉한 플레이어를 서버로 보내고 서버에서 해당 플레이어로 값을 전송하여 해당 콜백함수 에서 실행하는 방법과
2번째로는 클라이언트의 세션아이디를 저장해서 플레이어 인스턴스 시 마다 세션아이디를 저장하고 세션아이디 값이 일치하는 지 확인하는 방법이 있겠습니다.
